packages/purpose-process/src/services/purposeService.ts

```diff
@@ -811,3 +882,326 @@ const performUpdatePurpose = async (
+
+    const riskAnalysisDocument = await pdfGenerator.createRiskAnalysisDocument(
+      documentId,
+      purpose,
+      purposeVersion,
+      eserviceInfo,
+      tenantKind,
+      storeFile,
+      logger
+    );
+
+    const updatedPurposeVersion: PurposeVersion = {
+      ...purposeVersion,
+      state: purposeVersionState.active,
+      riskAnalysis: riskAnalysisDocument,
+      updatedAt: new Date(),
+      firstActivationAt: new Date(),
+    };
+
+    const updatedPurpose: Purpose = archiveOldVersion(
+      replacePurposeVersion(purpose, updatedPurposeVersion),
+      updatedPurposeVersion.id
+    );
+
+    return {
+      event: toCreateEventPurposeActivated({
+        purpose: updatedPurpose,
+        version,
+        correlationId,
+      }),
+      updatedPurposeVersion,
+    };
+  }
+
+  function activateFromSuspended(): {
+    event: CreateEvent<PurposeEvent>;
+    updatedPurposeVersion: PurposeVersion;
+  } {
+    function calcNewVersionState(state: {
+      suspendedByProducer?: boolean;
+      suspendedByConsumer?: boolean;
+    }): PurposeVersionState {
+      return match(state)
+        .with(
+          {
+            suspendedByConsumer: true,
+          },
+          () => purposeVersionState.suspended
+        )
+        .with(
+          {
+            suspendedByProducer: true,
+          },
+          () => purposeVersionState.suspended
+        )
+        .otherwise(() => purposeVersionState.active);
+    }
+
+    if (ownership === "PRODUCER" || ownership === "SELF_CONSUMER") {
+      const newState = calcNewVersionState({
+        suspendedByProducer: false,
+        suspendedByConsumer: purpose.suspendedByConsumer,
+      });
+      const updatedPurposeVersion: PurposeVersion = {
+        ...purposeVersion,
+        updatedAt: new Date(),
+        suspendedAt:
+          newState !== purposeVersionState.suspended
+            ? undefined
+            : purposeVersion.suspendedAt,
+        state: newState,
+      };
+      const updatedPurpose: Purpose = replacePurposeVersion(
+        purpose,
+        updatedPurposeVersion
+      );
+      return {
+        event: toCreateEventPurposeVersionUnsuspenedByProducer({
+          purpose: { ...updatedPurpose, suspendedByProducer: false },
+          versionId: purposeVersion.id,
+          version,
+          correlationId,
+        }),
+        updatedPurposeVersion,
+      };
+    } else {
+      const newState = calcNewVersionState({
+        suspendedByProducer: false,
+        suspendedByConsumer: purpose.suspendedByConsumer,
+      });
+      const updatedPurposeVersion: PurposeVersion = {
+        ...purposeVersion,
+        updatedAt: new Date(),
+        suspendedAt:
+          newState !== purposeVersionState.suspended
+            ? undefined
+            : purposeVersion.suspendedAt,
+        state: newState,
+      };
+      const updatedPurpose: Purpose = replacePurposeVersion(
+        purpose,
+        updatedPurposeVersion
+      );
```

Comment:
What are the differences between these lines and [the ones above](https://github.com/pagopa/interop-be-monorepo/pull/439/files#diff-95b4e078c342343517e86a98ea5650ede6c4ec6e5c9f4547dbfb44c8d6d22118R1078-R1094)?

Reply:
There is a typo that needs to be corrected, thanks for noticing!